lib/Elevate/Components/PackageRestore.pm

```diff
@@ -0,0 +1,106 @@
+package Elevate::Components::PackageRestore;
+
+=encoding utf-8
+
+=head1 NAME
+
+Elevate::Components::PackageRestore
+
+Handle restoring packages that get removed during elevate
+
+Before leapp:
+    Detect which packages in our list are installed and
+    store our findings.
+
+After leapp:
+    Reinstall any packages detected pre-leapp
+
+=cut
+
+use cPstrict;
+
+use Elevate::StageFile ();
+
+use Cpanel::Pkgr ();
+
+use File::Copy    ();
+use Log::Log4perl qw(:easy);
+
+use parent qw{Elevate::Components::Base};
+
+#
+# Set as a function for unit testing
+#
+sub _get_packages_to_check () {
+    return qw{
+      net-snmp
+    };
+}
+
+#
+# Set as a function for unit testing
+#
+sub _get_files_to_restore ($package) {
+
+    my $package_files_to_restore = {
+        'net-snmp' => [
```

Comment:
These aren't the only configurable files. Namely, the package also has `/etc/sysconfig/snmpd` and `/etc/sysconfig/snmptrapd` as ways to affect the arguments the services use to start the daemons. Should not those also be backed up?

Reply:
I wasn't aware if the files under sysconfig.  I can add those to the list.

Reply:
Rather than hardcoding this, it would be better to use `rpm -qc` to get this list or you could make `Elevate::RPM::_get_config_files` public and just call `$self->rpm->get_config_files( [ 'net-snmp' ] );` to get the list.

Reply:
That is a good idea.  I didn't know about `rpm -qc`.  That'd be much better.